src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -437,12 +437,14 @@ SHAMap::unshareNode(std::shared_ptr<Node> node, SHAMapNodeID const& nodeID)
 }
 
 SHAMapLeafNode*
-SHAMap::firstBelow(
+SHAMap::below(
     std::shared_ptr<SHAMapTreeNode> node,
     SharedPtrNodeStack& stack,
-    int branch) const
+    int branch,
+    std::tuple<int, std::function<bool(int)>, std::function<void(int&)>> const&
```

Comment:
I don't think packing all of the parameters together makes sense. This parameter pack is the loopParams, and so I think it makes sense to pack them together, since they are intimately related to each other. I originally wasn't going to pack them at all, but @seelabs suggested I pack them.

Reply:
I suggested we put them into a struct, but I also said I was OK with whatever solution cj decided on (and I still am).

Reply:
His suggestion packed them into a type to make one template parameter pack for all the template parameters. Here, they're not template parameters (despite being constants).

Reply:
I feel like we are straying into bike shedding territory here. What is the tangible issue with the way the code is right now? Readability, maintainability, something else?

Reply:
I think and hope the team wants to converge on a single code style for readability and maintenance, so "bikeshedding" is a necessary part of that process. Am I alone in that?

Reply:
So what is the single code style that should be used here then?